Assets/MixedRealityToolkit/Providers/BaseInputDeviceManager.cs

```diff
@@ -61,36 +183,168 @@ public abstract class BaseInputDeviceManager : BaseDataProvider<IMixedRealityInp
         /// <param name="useSpecificType">Only register pointers with a specific type.</param>
         protected virtual IMixedRealityPointer[] RequestPointers(SupportedControllerType controllerType, Handedness controllingHand)
         {
-            var pointers = new List<IMixedRealityPointer>();
-
-            if ((Service != null) &&
-                (InputSystemProfile != null) &&
-                InputSystemProfile.PointerProfile != null)
+            using (RequestPointersPerfMarker.Auto())
             {
-                for (int i = 0; i < InputSystemProfile.PointerProfile.PointerOptions.Length; i++)
+                var returnPointers = new List<IMixedRealityPointer>();
+
+                CleanActivePointers();
+
+                for (int i = 0; i < pointerConfigurations.Length; i++)
                 {
-                    var pointerProfile = InputSystemProfile.PointerProfile.PointerOptions[i];
+                    var option = pointerConfigurations[i].profile;
+                    if (option.ControllerType.HasFlag(controllerType) && option.Handedness.HasFlag(controllingHand))
+                    {
+                        IMixedRealityPointer requestedPointer = null;
+
+                        if (EnablePointerCache)
+                        {
+                            var pointerCache = pointerConfigurations[i].cache;
+                            while (pointerCache.Count > 0)
+                            {
+                                var p = pointerCache.Pop();
+                                var pointerComponent = p as MonoBehaviour;
+                                if (!UnityObjectExtensions.IsNull(pointerComponent))
+                                {
+                                    pointerComponent.gameObject.SetActive(true);
+
+                                    // We got pointer from cache, continue to next pointer option to review
+                                    requestedPointer = p;
+                                    break;
+                                }
+                            }
+                        }
+
+                        if (requestedPointer == null)
```

Comment:
should you only create this pointer if the hand and controller type match the current option? It seems like with this code, even if i have an articulated hand up, you will create all poitners for vr controllers and mouse controllers as well.

Reply:
Yes look at line 199
` if (option.ControllerType.HasFlag(controllerType) && option.Handedness.HasFlag(controllingHand))`

We have to pass this gate before even instantiating or accessing the cache for a pointer